### DISCUSSION TITLE
Cannot deploy app using custom domain/sub-domain in Vercel REST API

### DISCUSSION THREAD

**Discussion by ghubnab99** [vercel/community#3417] [Help]
### Summary

How can we specify a custom domain / sub-domain that the project should be deployed with using the create new deployment REST API?

POST: Create a new deployment: (https://vercel.com/docs/rest-api/endpoints#create-a-new-deployment)
https://api.vercel.com/v13/deployments

Currently we are able to create a new deployment using the REST API but it generates a URL like the below:
https://some-random-name.vercel.app/

instead we want the new deployment to have a URL like this:
https://sub-domain.custom-domain.com

After we connect our domain to our Vercel account and enable wildcards, can we specify, like a custom sub-domain in the deployment API params, to create the new deployment with the custom sub-domain?

Kindly guide us.
Thank you

### Steps to Reproduce

POST: Create a new deployment: (https://vercel.com/docs/rest-api/endpoints#create-a-new-deployment)
https://api.vercel.com/v13/deployments

{
  "gitSource": {
      "ref" : "master",
      "repoId": "66123400",
      "type": "github",
      "sha": "bd021512511"
      },
    "projectSettings": {
      "framework": "nextjs"
    },      
  "name": "nextjs-some-project"
}

**Comment by amyegan** [comment]
You can assign additional domains and subdomains to individual deployments by adding an `alias` to the request body. 

For a more permanent solution, you can use a different endpoint to [add a domain to a project](https://vercel.com/docs/rest-api/endpoints#add-a-domain-to-a-project) to it will automatically used for future deployments. With this approach, you can designate different subdomains for specific branches.

**Reply by ghubnab99** [reply]
Hi @amyegan,

You're truly a life saver for us. You've helped us once again.

We did try the "alias" param and it did work. But we have a query regarding this?

We used the "alias" param to create a new deployment using the REST API, and specified a custom domain: "car-rental.fktechno.com"
The deployment was successfully created
The website was also up and running at "car-rental.fktechno.com".
However, when we go to the "Domains" section of the project, there is only one single entry, and that of "nextjs-car-rental.vercel.app"
The entry for the custom domain "car-rental.fktechno.com" does not show up in the Domains section.

Moreover in the response of the create deployment REST API, along with other information we also get:
"aliasAssigned": false,

Why is this? 

If you could please advise

Thanks

**Reply by martinkz** [reply]
I'm having the same issue, the alias (my-alias.vercel.app) gets added, but is there a way to make it appear on the various Vercel admin dashboards as the main one? Instead it shows the random generated one as the primary .vercel.app domain.
